app/validators/checksum_validator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+class ChecksumValidator < ActiveModel::Validator
+  include Rails.application.routes.url_helpers
+  def validate(record)
+    @record = record
+    @uploaded_file = @record.file
+    @duvs_obj = DuplicateUploadVerificationService.new(@uploaded_file)
```

Comment:
@RudyOnRails you could memoize this in a private method for a cleaner look:
``` ruby
def duvs_object
  @duvs_object ||= DuplicateUploadVerificationService.new(@record.file)
end
```
Also, why not `verification_service` for the method/variable? It's a little more wordy, but would show up if you were searching for the words.